learners/setup.md

```diff
@@ -1,50 +1,96 @@
+[Python][python] is a popular language for scientific computing and a frequent choice for machine learning.
 
-Open a terminal (Mac/Linux) or Command Prompt (Windows) and run the following commands.
+Python version requirement: This workshop requires Python 3.11.9. Newer versions like 3.12 or 3.13 are not yet fully compatible with TensorFlow and may cause issues. Even Python 3.11.9 may have some edge cases, but it works well enough to be the default in Google Colab and is stable for the purposes of this workshop.
 
-1. Create a [virtual environment](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#create-and-use-virtual-environments) called `dl_workshop`:
+To install Python 3.11.9, go to the [official 3.11.9 downloads page](https://www.python.org/downloads/release/python-3119//). Choose the installer that matches your operating system (Windows, macOS, or Linux).
 
-::: spoiler
+Please set up your Python environment at least a day in advance of the workshop. If you run into issues with installation, contact the workshop organizers by email so you're ready to begin on time.
 
-### On Linux/macOs
+
+### Determine which `python` command to use for downstream setup steps
+
+Different systems and Python installations (e.g., Anaconda, Git Bash, system Python, Windows Store, etc.) may register different command names. This quick check helps identify which one points to Python 3.11.9 on your machine.
+
+Run the following in your terminal ([Git Bash recommended for Windows users](https://git-scm.com/downloads), Anaconda Prompt, or macOS/Linux shell):
 
 ```shell
-python3 -m venv dl_workshop
+python --version
+py --version
+python3 --version
 ```
 
-:::
+Use whichever one returns Python 3.11.9 for the rest of the setup steps.
 
-::: spoiler
+Example output:
 
-### On Windows
+```output
+$ python --version
+Python 3.11.9
+
+$ py --version
+Python 3.13.2
+
+$ python3 --version
+Python was not found...
+```
+In this case, use python throughout the remainder of the instructions.
+
+If none of the commands return Python 3.11.9:
+
+- Download and install Python 3.11.9. On Windows, be sure to check "Add Python to PATH" during installation
+- If you're on Windows using Anaconda Prompt, try using [Git Bash](https://git-scm.com/downloads) instead.  
+- Then re-run the checks above in a new terminal window
+
+If you're still stuck, ask the workshop organizers for help before proceeding.
+
+## 3. Configure virtual environment
+
+Open a terminal and run the following commands.
+
+1. Create a [virtual environment](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#create-and-use-virtual-environments) called `venv` using the "venv" command:
 
 ```shell
-py -m venv dl_workshop
+# Use python3 or py instead if one of them points to 3.11.9.
+python -m venv venv # 1st "venv" is commmand, 2nd venv is name of the virtual environment / folder
```

Comment:
The guide linked above uses `.venv` as environment folder, which has become best practice. The downside is (on purpose) that the folder is invisible by default, but I think I would prefer sticking with the instructions to avoid confusion.
```suggestion
python -m venv .venv
```